src/bin/ion/commands/dump.rs

```diff
@@ -95,18 +90,30 @@ pub(crate) fn write_in_format(
     format: &str,
     count: Option<usize>,
 ) -> IonResult<usize> {
-    match format {
+    // XXX: The text formats below each have additional logic to append a newline because the
+    //      ion-rs writer doesn't handle this automatically like it should.
```

Comment:
🗺️ Addresses #36. The real fix still needs to happen upstream in `ion-rs`. See amazon-ion/ion-rust#437.